crates/sparrow-compiler/tests/compiler_golden_tests.rs

```diff
@@ -614,16 +614,14 @@ async fn test_incompatible_err() {
     .await;
 
     insta::assert_snapshot!(error, @r###"
-    error[E0010]: Invalid argument type(s)
+    error[E0015]: Incompatible argument types
       --> Query:1:13
       |
     1 | Sent.amount == "hello"
       | ----------- ^^ ------- Type: string
       | |           |
-      | |           Invalid types for call to 'eq'
+      | |           Incompatible types for call to 'eq'
       | Type: f64
-      |
-      = Expected 'any'
     "###);
```

Comment:
@kerinin @bjchambers Thoughts on this change? 

When two types to a function are incompatible (cannot be promoted to one another), this error will take precedence over the `InvalidArgument` error. 

Reasoning: when doing the `get` function it didn't make sense to see 

```
key: i64, Map.key: string
= Expected 'key'
```

because both of these were of the `key` type, they just weren't compatible with each other.